cmd/kpod/create.go

```diff
@@ -115,46 +135,765 @@ var createCommand = cli.Command{
+			Access: "rwm",
+		},
+		{
+			Allow:  true,
+			Type:   "c",
+			Major:  iPtr(5),
+			Minor:  iPtr(0),
+			Access: "rwm",
+		},
+		{
+			Allow:  true,
+			Type:   "c",
+			Major:  iPtr(5),
+			Minor:  iPtr(1),
+			Access: "rwm",
+		},
+		{
+			Allow:  false,
+			Type:   "c",
+			Major:  iPtr(10),
+			Minor:  iPtr(229),
+			Access: "rwm",
+		},
+	}
+}
+
+func defaultCapabilities() []string {
+	return []string{
+		"CAP_CHOWN",
+		"CAP_DAC_OVERRIDE",
+		"CAP_FSETID",
+		"CAP_FOWNER",
+		"CAP_MKNOD",
+		"CAP_NET_RAW",
+		"CAP_SETGID",
+		"CAP_SETUID",
+		"CAP_SETFCAP",
+		"CAP_SETPCAP",
+		"CAP_NET_BIND_SERVICE",
+		"CAP_SYS_CHROOT",
+		"CAP_KILL",
+		"CAP_AUDIT_WRITE",
+	}
+}
+
+func (c *createConfig) GetDefaultLinuxSpec() spec.Spec {
+	s := spec.Spec{
+		Version: spec.Version,
+		Root:    &spec.Root{},
+	}
+	s.Annotations = c.GetAnnotations()
+	s.Mounts = c.GetDefaultMounts()
+	s.Process = &spec.Process{
+		Capabilities: &spec.LinuxCapabilities{
+			Bounding:    defaultCapabilities(),
+			Permitted:   defaultCapabilities(),
+			Inheritable: defaultCapabilities(),
+			Effective:   defaultCapabilities(),
+		},
+	}
+	s.Linux = &spec.Linux{
+		MaskedPaths: []string{
+			"/proc/kcore",
+			"/proc/latency_stats",
+			"/proc/timer_list",
+			"/proc/timer_stats",
+			"/proc/sched_debug",
+		},
+		ReadonlyPaths: []string{
+			"/proc/asound",
+			"/proc/bus",
+			"/proc/fs",
+			"/proc/irq",
+			"/proc/sys",
+			"/proc/sysrq-trigger",
+		},
+		Namespaces: []spec.LinuxNamespace{
+			{Type: "mount"},
+			{Type: "network"},
+			{Type: "uts"},
+			{Type: "pid"},
+			{Type: "ipc"},
+		},
+		Devices: []spec.LinuxDevice{},
+		Resources: &spec.LinuxResources{
+			Devices: c.GetDefaultDevices(),
+		},
+	}
+
+	return s
+}
+
+func (c *createConfig) GetAnnotations() map[string]string {
+	a := getDefaultAnnotations()
+	// TODO
+	// Which annotations do we want added by default
+	if c.tty {
+		a["io.kubernetes.cri-o.TTY"] = "true"
+	}
+	return a
+}
+
+func getDefaultAnnotations() map[string]string {
+	var a map[string]string
+	a = make(map[string]string)
+	a[ann.Annotations] = ""
+	a[ann.ContainerID] = ""
+	a[ann.ContainerName] = ""
+	a[ann.ContainerType] = ""
+	a[ann.Created] = ""
+	a[ann.HostName] = ""
+	a[ann.IP] = ""
+	a[ann.Image] = ""
+	a[ann.ImageName] = ""
+	a[ann.ImageRef] = ""
+	a[ann.KubeName] = ""
+	a[ann.Labels] = ""
+	a[ann.LogPath] = ""
+	a[ann.Metadata] = ""
+	a[ann.Name] = ""
+	a[ann.PrivilegedRuntime] = ""
+	a[ann.ResolvPath] = ""
+	a[ann.HostnamePath] = ""
+	a[ann.SandboxID] = ""
+	a[ann.SandboxName] = ""
+	a[ann.ShmPath] = ""
+	a[ann.MountPoint] = ""
+	a[ann.TrustedSandbox] = ""
+	a[ann.TTY] = "false"
+	a[ann.Stdin] = ""
+	a[ann.StdinOnce] = ""
+	a[ann.Volumes] = ""
+
+	return a
+}
+
+//GetTmpfsMounts takes user provided input for bind mounts and creates Mount structs
+func (c *createConfig) GetVolumeMounts() []spec.Mount {
+	var m []spec.Mount
+	var options []string
+	for _, i := range c.volumes {
+		spliti := strings.Split(i, ":")
```

Comment:
Need to handle :Z :z :ro

Reply:
It handles :Z,ro  .  I believe the format is :OPTION,OPTION,... right?

Reply:
Handling of :Z actually involves running SELinux calls to set the SELinux labels.

Reply:
We don't have SELinux support in libpod right now, but it wouldn't be hard to add.

Reply:
I think the relabeling needs to happen here, though, I don't have enough information in libpod to handle it.

Reply:
I'll add a comment here for us to handle selinux options here.  And follow up with you.